From 6ec3c02ed2a1e528830c921e0722e6f35c7c60bb Mon Sep 17 00:00:00 2001
From: Joseph Guo <qijian.guo@nxp.com>
Date: Thu, 17 Apr 2025 15:09:42 +0900
Subject: [PATCH 04/56] LF-15223 arm64: dts: imx8mp-frdm: enable BOE WXGA LVDS
 panel

Enable BOE WXGA LVDS panel for FRDM-iMX8MP board.
Add dts files for LVDS0, LVDS1

Signed-off-by: Joseph Guo <qijian.guo@nxp.com>
Acked-by: Liu Ying <victor.liu@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |  1 +
 .../imx8mp-frdm-boe-wuxga-lvds0-panel.dts     | 76 ++++++++++++++++++
 .../imx8mp-frdm-boe-wuxga-lvds1-panel.dts     | 78 +++++++++++++++++++
 3 files changed, 155 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-frdm-boe-wuxga-lvds0-panel.dts
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-frdm-boe-wuxga-lvds1-panel.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 5717d4964632..76b8306206e3 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -257,6 +257,7 @@ dtb-$(CONFIG_ARCH_MXC) += imx8mp-evk-revb4-rpmsg.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-evk-revb4-rpmsg-lpv.dtb
 
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-frdm.dtb
+dtb-$(CONFIG_ARCH_MXC) += imx8mp-frdm-boe-wuxga-lvds0-panel.dtb imx8mp-frdm-boe-wuxga-lvds1-panel.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-frdm-os08a20.dtb imx8mp-frdm-os08a20-dual.dtb
 
 dtb-$(CONFIG_ARCH_MXC) += imx8mq-evk.dtb imx8mq-evk-rpmsg.dtb
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-frdm-boe-wuxga-lvds0-panel.dts b/arch/arm64/boot/dts/freescale/imx8mp-frdm-boe-wuxga-lvds0-panel.dts
new file mode 100644
index 000000000000..4dd5d14428c1
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-frdm-boe-wuxga-lvds0-panel.dts
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2025 NXP
+ */
+
+#include "imx8mp-frdm.dts"
+
+/ {
+	panel-lvds0 {
+		compatible = "boe,ev121wxm-n10-1850";
+		backlight = <&backlight_lvds0>;
+		power-supply = <&reg_lvds_vcc>;
+
+		port {
+			panel_lvds_in: endpoint {
+				remote-endpoint = <&lvds_out>;
+			};
+		};
+	};
+
+	backlight_lvds0: backlight-lvds0 {
+		compatible = "pwm-backlight";
+		enable-gpios = <&pca6416_a 8 GPIO_ACTIVE_HIGH>;
+		pwms = <&pwm3 0 100000 0>;
+		power-supply = <&reg_vdd_12v>;
+		brightness-levels = <0 100>;
+		num-interpolated-steps = <100>;
+		default-brightness-level = <80>;
+	};
+};
+
+&i2c2 {
+	touchscreen@2a {
+		compatible = "eeti,exc80h60";
+		reg = <0x2a>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_lvds_ctp_int_0>;
+		interrupt-parent = <&gpio4>;
+		interrupts = <28 IRQ_TYPE_LEVEL_LOW>;
+		reset-gpios = <&pca6416_a 5 GPIO_ACTIVE_HIGH>;
+		touchscreen-size-y = <4096>;
+		touchscreen-size-x = <4096>;
+	};
+};
+
+&lcdif2 {
+	status = "okay";
+};
+
+&ldb {
+	status = "okay";
+
+	lvds-channel@0 {
+		status = "okay";
+
+		port@1 {
+			reg = <1>;
+
+			lvds_out: endpoint {
+				remote-endpoint = <&panel_lvds_in>;
+			};
+		};
+	};
+};
+
+&ldb_phy {
+	status = "okay";
+};
+
+&iomuxc {
+	pinctrl_lvds_ctp_int_0: lvds_ctp_int_grp_0 {
+		fsl,pins = <
+			MX8MP_IOMUXC_SAI3_RXFS__GPIO4_IO28		  0x31e
+		>;
+	};
+};
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-frdm-boe-wuxga-lvds1-panel.dts b/arch/arm64/boot/dts/freescale/imx8mp-frdm-boe-wuxga-lvds1-panel.dts
new file mode 100644
index 000000000000..ed9bce0373eb
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-frdm-boe-wuxga-lvds1-panel.dts
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2025 NXP
+ */
+
+#include "imx8mp-frdm.dts"
+
+/ {
+	panel-lvds1 {
+		compatible = "boe,ev121wxm-n10-1850";
+		backlight = <&backlight_lvds1>;
+		power-supply = <&reg_lvds_vcc>;
+
+		port {
+			panel_lvds_in: endpoint {
+				remote-endpoint = <&lvds_out>;
+			};
+		};
+	};
+
+	backlight_lvds1: backlight-lvds1 {
+		compatible = "pwm-backlight";
+		enable-gpios = <&pca6416_a 9 GPIO_ACTIVE_HIGH>;
+		pwms = <&pwm2 0 100000 0>;
+		power-supply = <&reg_vdd_12v>;
+		brightness-levels = <0 100>;
+		num-interpolated-steps = <100>;
+		default-brightness-level = <80>;
+	};
+};
+
+&i2c3 {
+	touchscreen@2a {
+		compatible = "eeti,exc80h60";
+		reg = <0x2a>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_lvds_ctp_int_1>;
+		interrupt-parent = <&gpio4>;
+		interrupts = <29 IRQ_TYPE_LEVEL_LOW>;
+		reset-gpios = <&pca6416_a 6 GPIO_ACTIVE_HIGH>;
+		touchscreen-size-y = <4096>;
+		touchscreen-size-x = <4096>;
+	};
+};
+
+&lcdif2 {
+	status = "okay";
+};
+
+&ldb {
+	status = "okay";
+
+	lvds-channel@1 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		status = "okay";
+
+		port@1 {
+			reg = <1>;
+
+			lvds_out: endpoint {
+				remote-endpoint = <&panel_lvds_in>;
+			};
+		};
+	};
+};
+
+&ldb_phy {
+	status = "okay";
+};
+
+&iomuxc {
+	pinctrl_lvds_ctp_int_1: lvds_ctp_int_grp_1 {
+		fsl,pins = <
+			MX8MP_IOMUXC_SAI3_RXC__GPIO4_IO29		  0x31e
+		>;
+	};
+};
-- 
2.34.1

